scripts/smithy.lic

```diff
@@ -0,0 +1,139 @@
+=begin
+
+  Smithy service helper script
+
+  This script will request multiple services from the smithy. It tries to be
+  strict about what it will accept for input and to verify that a blacksmith
+  NPC is present before requesting services.
+
+  Usage:
+
+  ;smithy <count> <crit, damage, or sighting>
+
+  author: Kragdruk
+  version: 1.0
+  tags: util, ccf, duskruin, festival, merchant
+
+=end
+
+module Smithy
+  SERVICE_TYPES = %w[critical damage sighting]
+
+  def self.determine_service_type_from(arg)
+    SERVICE_TYPES.find { |t| t.start_with? arg.downcase }
+  end
+
+  BLACKSMITH_DESCRIPTIONS = [
+    "old dwarven blacksmith",
+  ]
+
+  def self.blacksmith
+    GameObj.npcs.find do |npc|
+      BLACKSMITH_DESCRIPTIONS.include? npc.name
+    end
+  end
+
+  def self.print_usage
+    respond
+    respond "  Usage:"
+    respond
+    respond "  #{$clean_lich_char}#{Script.current.name} <count> <crit, damage, or sighting>"
+    respond
+  end
+
+  def self.perform_services(opts)
+    opts.fetch(:count).times do
+      perform_service_request(opts.fetch(:service))
+      perform_service_confirmation
+    end
+  end
+
+  def self.perform_service_request(service_type)
+    if result = dothistimeout("ask ##{blacksmith.id} about #{service_type}", 5, /Modified \w+ Type: #{service_type}/i)
+      result
+    else
+      respond
+      respond "  ERROR: doesn't seem that we were able to ask the blacksmith for work"
+      respond
+      exit
```

Comment:
I considered making a `fatal` helper method for this repeated idiom and/or switching to exceptions, but didn't feel strongly enough either.

Reply:
I prefer exceptions, because you get free stacktraces to help knowledgeable people debug